src/librbd/io/ImageRequestWQ.cc

```diff
@@ -423,6 +442,23 @@ void *ImageRequestWQ::_void_dequeue() {
       }
     } else if (m_require_lock_on_read) {
       return nullptr;
+    } else {
+      if (m_image_ctx.max_read_iops) {
```

Comment:
Seems like you should be able to combine all of this duplicate logic by moving the throttling to the top or using a helper function.